### PR TITLE
Enable gometalinter on tests and fix static analysis issues #631

### LINKED ISSUE #631: Enable gometalinter for test files
# Problem

Currently, we do not lint Go test files (`*_test.go`) but we should. Test files, just like the normal ones, should be reliable and bug-free.

# Implementation

Add `"Test": true` property to `.gometalinter.json`. It also requires fixing all lint errors and warning that will be found.

# Acceptance Criteria

1. Go test files are linted,
1. `make lint` passes.

### PR DESCRIPTION
**Pull request by alexkohler**
Enable Gometalinter on tests and fix static analysis issues. 

Fixes #631

### CODE REVIEW

File: geth/jail/internal/loop/loop_test.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (s *LoopSuite) SetupTest() {
 
 	ctx, cancel := context.WithCancel(context.Background())
 	s.cancel = cancel
-	go s.loop.Run(ctx)
+	go func() { _ = s.loop.Run(ctx) }()
```
Comment (b00ris) [inline geth/jail/internal/loop/loop_test.go]: I think nolint tag looks more pretty

File: geth/mailservice/mailservice_api_test.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ func TestRequestMessagesFailures(t *testing.T) {
 	nodeA, nodeErr := node.New(&node.Config{NoUSB: true})
 	require.NoError(t, nodeErr)
 	require.NoError(t, nodeA.Start())
-	defer nodeA.Stop()
+	defer func() { _ = nodeA.Stop() }()
```
Comment (b00ris) [inline geth/mailservice/mailservice_api_test.go]: Same

File: t/e2e/whisper/whisper_mailbox_test.go [hunk 3]
```diff
@@ -213,7 +213,7 @@ func (s *WhisperMailboxSuite) TestRequestMessagesInGroupChat() {
 	messages := s.getMessagesByMessageFilterID(bobRPCClient, bobMessageFilterID)
 	s.Require().Equal(1, len(messages))
 	bobGroupChatData := groupChatParams{}
-	bobGroupChatData.Decode(messages[0]["payload"].(string))
+	_ = bobGroupChatData.Decode(messages[0]["payload"].(string))
```
Comment (b00ris) [inline t/e2e/whisper/whisper_mailbox_test.go]: It would be better to add error check here

File: t/e2e/whisper/whisper_mailbox_test.go [hunk 4]
```diff
@@ -228,7 +228,7 @@ func (s *WhisperMailboxSuite) TestRequestMessagesInGroupChat() {
 	messages = s.getMessagesByMessageFilterID(charlieRPCClient, charlieMessageFilterID)
 	s.Require().Equal(1, len(messages))
 	charlieGroupChatData := groupChatParams{}
-	charlieGroupChatData.Decode(messages[0]["payload"].(string))
+	_ = charlieGroupChatData.Decode(messages[0]["payload"].(string))
```
Comment (b00ris) [inline t/e2e/whisper/whisper_mailbox_test.go]: It would be better to add error check here

File: metrics/node/subscribe_test.go [hunk 5]
```diff
@@ -19,7 +19,7 @@ func TestSubscribeServerEvents(t *testing.T) {
 	require.NoError(t, err)
 	err = node.Start()
 	require.NoError(t, err)
-	defer node.Stop()
+	defer func() { _ = node.Stop() }()
```
Comment (b00ris) [inline metrics/node/subscribe_test.go]: Same

File: geth/jail/cell_test.go [hunk 6]
```diff
@@ -237,7 +237,7 @@ func (s *CellTestSuite) TestCellCallAsync() {
 	fn, err := s.cell.Get("testCallAsync")
 	s.NoError(err)
 
-	s.cell.CallAsync(fn, "success")
+	_ = s.cell.CallAsync(fn, "success")
```
Comment (b00ris) [inline geth/jail/cell_test.go]: It would be better to add error check here

**Review by b00ris — CHANGES_REQUESTED** [review]
Thank you for commiting to status! PR's ok, but i think you could do it a little bit better

**Comment by alexkohler** [comment]
@b00ris Thanks for the input! Got in a rush towards the end of lunch haha. I'll add nolint directives and error checks where appropriate.

**Review by b00ris — APPROVED** [review]
LGTM

File: t/e2e/whisper/whisper_mailbox_test.go [hunk 7]
```diff
@@ -323,7 +325,7 @@ func (s *WhisperMailboxSuite) startBackend(name string) (*api.StatusBackend, fun
 		s.NoError(err)
 		<-backendStopped
 		s.False(backend.IsNodeRunning())
-		_ = os.RemoveAll(datadir)
+		os.RemoveAll(datadir) //nolint: errcheck
```
Comment (dshulyak) [inline t/e2e/whisper/whisper_mailbox_test.go]: why not to check error if you are changing this line anyway?

File: metrics/node/subscribe_test.go [hunk 8]
```diff
@@ -19,7 +19,7 @@ func TestSubscribeServerEvents(t *testing.T) {
 	require.NoError(t, err)
 	err = node.Start()
 	require.NoError(t, err)
-	defer func() { _ = node.Stop() }()
+	defer node.Stop() //nolint: errcheck
```
Comment (dshulyak) [inline metrics/node/subscribe_test.go]: you could do func() {t.NoError}() for every case

**Review by adambabik — CHANGES_REQUESTED** [review]
Thanks @alexkohler for the PR! 

Please take a look at @dshulyak comments. There are good suggestions how to avoid `//nolint` as we deal with test files here and using `s.Require().NoError()` is possible.

File: metrics/node/subscribe_test.go (hunk 8 above)
Comment (alexkohler) [reply metrics/node/subscribe_test.go]: I'm not sure I understand your comment - you mean something along the lines of:
```
defer func() {
    err := node.Stop()
    require.NoError(err)
}
```

File: geth/jail/cell_test.go [hunk 9]
```diff
@@ -151,7 +151,10 @@ func (s *CellTestSuite) TestCellFetchErrorRace() {
 	// Find a free port in localhost
 	freeportListener, err := net.Listen("tcp", "127.0.0.1:0")
 	s.Require().NoError(err)
-	defer freeportListener.Close()
+	defer func() {
+		err := freeportListener.Close()
+		s.NoError(err)
```
Comment (dshulyak) [inline geth/jail/cell_test.go:156]: not a big deal, but why not to do check error on the same line

File: geth/jail/internal/loop/loop_test.go [hunk 10]
```diff
@@ -53,7 +53,7 @@ func (s *LoopSuite) SetupTest() {
 
 	ctx, cancel := context.WithCancel(context.Background())
 	s.cancel = cancel
-	go s.loop.Run(ctx)
+	go s.loop.Run(ctx) //nolint: errcheck
```
Comment (dshulyak) [inline geth/jail/internal/loop/loop_test.go]: it will be useful to know an error if loop will crush

File: geth/jail/cell_test.go (hunk 9 above)
Comment (alexkohler) [reply geth/jail/cell_test.go:156]: I can, it looked like the error check was on a separate line in the majority of the codebase - if you feel strongly I can change this for all the error checks in this PR

File: geth/jail/internal/loop/loop_test.go (hunk 10 above)
Comment (alexkohler) [reply geth/jail/internal/loop/loop_test.go]: I'm not 100% familiar with these tests, but it looks like in all of these tests we're canceling the context at the end, which will then always returns a `context.Canceled` error here. I've added a check - please let me know if this isn't a reasonable assumption.

**Review by adambabik — APPROVED** [review]

**Pull request merged**